riscv-runtime/src/io.rs

```diff
@@ -51,6 +141,16 @@ pub fn write_slice(fd: u32, data: &[u8]) {
 use serde::de::DeserializeOwned;
 use serde::Serialize;
 
+pub fn read_stdin<T: DeserializeOwned>() -> T {
```

Comment:
idk about stdin... i think id prefer something more explicit, specially because you could also read stdin from the other method, and it will not behave the same?

Reply:
add_memory_input / read_memory_input ?

Reply:
Or use the `ProverDataReader` iterator directly...

I don't like this stateful interface.

Reply:
From the user's perspective, this is over complicated and just not good enough. In other zkVMs the user just calls "read", do we really want to force them to import some random type ProverDataReader just to read a T?